packages/p2p/src/components/my-ads/buy-ad-payment-methods-list.jsx

```diff
@@ -0,0 +1,202 @@
+import * as React from 'react';
+import { Formik, Field } from 'formik';
+import { observer } from 'mobx-react-lite';
+import { Autocomplete, Icon } from '@deriv/components';
+import { useStores } from 'Stores';
+import { localize } from 'Components/i18next';
+import PropTypes from 'prop-types';
+
+const BuyAdPaymentMethodsList = ({ selected_methods, setSelectedMethods }) => {
+    const { my_ads_store, my_profile_store } = useStores();
+    const [selected_edit_method, setSelectedEditMethod] = React.useState();
+    const [payment_methods_list, setPaymentMethodsList] = React.useState(
+        my_profile_store.payment_methods_list.filter(({ value }) => !selected_methods.includes(value))
+    );
+
+    React.useEffect(() => {
+        setPaymentMethodsList(
+            my_profile_store.payment_methods_list.filter(({ value }) => !selected_methods.includes(value))
+        );
+    }, [selected_methods]);
+
+    const onClickDeletePaymentMethodItem = value => {
+        if (value) {
+            my_ads_store.payment_method_names = my_ads_store.payment_method_names.filter(
+                payment_method_id => payment_method_id !== value
+            );
+            setSelectedMethods(selected_methods.filter(i => i !== value));
+            setPaymentMethodsList([
+                ...payment_methods_list,
+                {
+                    value,
+                    text: my_profile_store.getPaymentMethodDisplayName(value),
+                },
+            ]);
+        }
+    };
+
+    const onEditPaymentMethodItem = (value, index) => {
+        if (value && !my_ads_store.payment_method_names.includes(value)) {
+            const edited_methods = selected_methods.slice();
+            edited_methods[index] = value;
+            my_ads_store.payment_method_names[index] = value;
+            setSelectedMethods(edited_methods);
+            setPaymentMethodsList([
+                ...payment_methods_list.filter(payment_method => payment_method.value !== value),
+                selected_edit_method,
+            ]);
+        }
+    };
+
+    const onClickPaymentMethodItem = value => {
+        if (value && !my_ads_store.payment_method_names.includes(value)) {
+            if (my_ads_store.payment_method_names.length < 3) {
+                my_ads_store.payment_method_names.push(value);
+                setSelectedMethods([...selected_methods, value]);
+                setPaymentMethodsList(payment_methods_list.filter(payment_method => payment_method.value !== value));
+            }
+        }
+    };
+
+    const checkValidPaymentMethod = payment_method_text => {
+        let match = false;
+        let id;
+        my_profile_store.payment_methods_list.map(({ value, text }) => {
+            if (text === payment_method_text) {
+                match = true;
+                id = value;
+            }
+        });
```

Comment:
Errr would be better if you use the `Array.find()` then returning the value. That way you dont need the temporary variables of `match` and `id`